### PR TITLE
Add dotfile repo support

### LINKED ISSUE #5198: Epic: Personalize workspaces from a user-configurable dotfiles repository
## Proposal

Provision the environment from a dotfiles repository, e.g. https://gitlab.com/dnsmichi/dotfiles 

May need a special entry point in the .gitpod.yml similar to init/tasks: https://www.gitpod.io/docs/config-gitpod-file 

## Context

Question from @eddiejaoude in the stream at https://www.youtube.com/watch?v=tOrr4r_pNQc 

GitHub Codespaces provides this feature: https://docs.github.com/en/codespaces/customizing-your-codespace/personalizing-codespaces-for-your-account#dotfiles 

## Possible Workarounds

Manual commands in .gitpod.yml and automate the setup of a dotfiles repository. 

---
More [details in the RFC](https://gitpod.notion.site/dotfiles-in-Gitpod-workspaces-b46b8723e9fe4efdbede72daa311961f)

### PR DESCRIPTION
**Pull request by csweichel**
## Description
This PR implements dotfile support, whereby a user can provide a "dotfile repo" which gets cloned during workspace startup. We implement exactly what's described in the [dotfile RFC](https://github.com/gitpod-io/gitpod/issues/5198).

## Related Issue(s)

Fixes #5198

## How to test
1. Go to the [preferences](https://cw-dotfiles.staging.gitpod-dev.com/preferences) and enter a dotfile repo (e.g. `https://github.com/Bash-it/bash-it.git`)
2. Start any workspace and notice how the terminal looks different than what you're used to
3. Inspect `$HOME/.dotfiles`
4. Check out the logs of the dotfile installation process: `cat ~/.dotfiles.log`

## Release Notes

```release-note
Add dotfile repo support
```

## Documentation

Will add an issue once it's clear how this feature will work and if it should land.

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **filiptronicek**, **gtsiolis** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @filiptronicek @gtsiolis` in a comment when ready.

Associated issue: *#5198*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by JanKoehnlein** [comment]
Awesome work, thanks!

**Comment by axonasif** [comment]
Hello @csweichel. I have a suggestion, it is about using the kernel provided `overlayfs` to overlay the cloned `/home/.dotfiles` on top of `/home/gitpod` as the lower dir.

Apart from that, I'm eagarly waiting for the dotfiles support to arrive!

**Comment by csweichel** [comment]
> Hello @csweichel. I have a suggestion, it is about using the kernel provided `overlayfs` to overlay the cloned `/home/.dotfiles` on top of `/home/gitpod` as the lower dir instead of using symlinks.
> 
> Apart from that, I'm eagarly waiting for the dotfiles support to arrive!

@axonasif Thank you for the feedback. I'm not sure I understand the use-case though; instead it strikes me that that approach comes with some drawbacks which make it unfavourable imho:
- the overlay would affect the entire `$HOME` directory and it's unclear what the meaning of the lowerdir would be then
- akin to the former point, due to the symlinks changes made to the files can be committed back to the repo, with the overlay that would not be so easy
- the filesystem setup in a workspace is already dangerously complex (shiftfs, fuse-overlayfs, idmapped mounts in the future), and I fret to add another layer to the mix.

**Comment by jldec** [comment]
I tried this PR with my slightly naive dotfiles repo which just tries to set an alias using [`.bash_profile`](https://github.com/jldec/dotfiles/blob/main/.bash_profile)
See https://github.com/jldec/dotfiles

I noticed from the .dotfiles.log that we are symlinking the `.git` directory (bug) and also the `README.md` file (bug?), but not my `.bash_profile` (maybe because there is already a .bash_profile in which case the log should show the failure)

```
gitpod ~ $ cat .dotfiles.log
Cloning into '/home/gitpod/.dotfiles'...
# installation script candidate /home/gitpod/.dotfiles/install.sh is not available
# installation script candidate /home/gitpod/.dotfiles/install is not available
# installation script candidate /home/gitpod/.dotfiles/bootstrap.sh is not available
# installation script candidate /home/gitpod/.dotfiles/bootstrap is not available
# installation script candidate /home/gitpod/.dotfiles/script/bootstrap is not available
# installation script candidate /home/gitpod/.dotfiles/setup.sh is not available
# installation script candidate /home/gitpod/.dotfiles/setup is not available
# installation script candidate /home/gitpod/.dotfiles/script/setup is not available
# echo linking /home/gitpod/.dotfiles/.git/HEAD -> /home/gitpod/.git/HEAD
# echo linking /home/gitpod/.dotfiles/.git/config -> /home/gitpod/.git/config
# echo linking /home/gitpod/.dotfiles/.git/description -> /home/gitpod/.git/description
# echo linking /home/gitpod/.dotfiles/.git/hooks/applypatch-msg.sample -> /home/gitpod/.git/hooks/applypatch-msg.sample
# echo linking /home/gitpod/.dotfiles/.git/hooks/commit-msg.sample -> /home/gitpod/.git/hooks/commit-msg.sample
# echo linking /home/gitpod/.dotfiles/.git/hooks/fsmonitor-watchman.sample -> /home/gitpod/.git/hooks/fsmonitor-watchman.sample
# echo linking /home/gitpod/.dotfiles/.git/hooks/post-update.sample -> /home/gitpod/.git/hooks/post-update.sample
# echo linking /home/gitpod/.dotfiles/.git/hooks/pre-applypatch.sample -> /home/gitpod/.git/hooks/pre-applypatch.sample
# echo linking /home/gitpod/.dotfiles/.git/hooks/pre-commit.sample -> /home/gitpod/.git/hooks/pre-commit.sample
# echo linking /home/gitpod/.dotfiles/.git/hooks/pre-merge-commit.sample -> /home/gitpod/.git/hooks/pre-merge-commit.sample
# echo linking /home/gitpod/.dotfiles/.git/hooks/pre-push.sample -> /home/gitpod/.git/hooks/pre-push.sample
# echo linking /home/gitpod/.dotfiles/.git/hooks/pre-rebase.sample -> /home/gitpod/.git/hooks/pre-rebase.sample
# echo linking /home/gitpod/.dotfiles/.git/hooks/pre-receive.sample -> /home/gitpod/.git/hooks/pre-receive.sample
# echo linking /home/gitpod/.dotfiles/.git/hooks/prepare-commit-msg.sample -> /home/gitpod/.git/hooks/prepare-commit-msg.sample
# echo linking /home/gitpod/.dotfiles/.git/hooks/push-to-checkout.sample -> /home/gitpod/.git/hooks/push-to-checkout.sample
# echo linking /home/gitpod/.dotfiles/.git/hooks/update.sample -> /home/gitpod/.git/hooks/update.sample
# echo linking /home/gitpod/.dotfiles/.git/index -> /home/gitpod/.git/index
# echo linking /home/gitpod/.dotfiles/.git/info/exclude -> /home/gitpod/.git/info/exclude
# echo linking /home/gitpod/.dotfiles/.git/logs/HEAD -> /home/gitpod/.git/logs/HEAD
# echo linking /home/gitpod/.dotfiles/.git/logs/refs/heads/main -> /home/gitpod/.git/logs/refs/heads/main
# echo linking /home/gitpod/.dotfiles/.git/logs/refs/remotes/origin/HEAD -> /home/gitpod/.git/logs/refs/remotes/origin/HEAD
# echo linking /home/gitpod/.dotfiles/.git/objects/pack/pack-9410ec61f4941fa5eb205496cdd147458b33ffbf.idx -> /home/gitpod/.git/objects/pack/pack-9410ec61f4941fa5eb205496cdd147458b33ffbf.idx
# echo linking /home/gitpod/.dotfiles/.git/objects/pack/pack-9410ec61f4941fa5eb205496cdd147458b33ffbf.pack -> /home/gitpod/.git/objects/pack/pack-9410ec61f4941fa5eb205496cdd147458b33ffbf.pack
# echo linking /home/gitpod/.dotfiles/.git/packed-refs -> /home/gitpod/.git/packed-refs
# echo linking /home/gitpod/.dotfiles/.git/refs/heads/main -> /home/gitpod/.git/refs/heads/main
# echo linking /home/gitpod/.dotfiles/.git/refs/remotes/origin/HEAD -> /home/gitpod/.git/refs/remotes/origin/HEAD
# echo linking /home/gitpod/.dotfiles/.git/shallow -> /home/gitpod/.git/shallow
# echo linking /home/gitpod/.dotfiles/README.md -> /home/gitpod/README.md
```

Beginner Q: If I just wanted to inject aliases with a dotfile, what would be the best way to do that, without overwriting the existing .bashrc in the workspace? Do I need an install script even for something as basic as this?

note: GitHub codespaces does a symlink:  
`.bash_profile -> /workspaces/.codespaces/.persistedshare/dotfiles/.bash_profile`
But the .bash_profile does not appear to be run in each codespaces terminal either. I think this is expected because VS Code terminal are not login shells.

**Comment by jldec** [comment]
Here's a suggested layout and copy for shipping this PR as Beta, while we work on the repo selection UI (issue #7385).  
We should also add a link to the docs as soon as those are available. (gitpod-io/website#1407)
cc: @gtsiolis 

Screenshot 2021-12-29 at 17 34 37

---
Note `max-w-2xl` for the outer container, and `max-w-xl` for the input.

Screenshot 2021-12-29 at 17 33 44

**Comment by filiptronicek** [comment]
Random idea: on new accounts, we could check their connected git provider (all of them maybe even) and if any turn up with a dotfiles repo, we could have it be there as a default option, so that it makes it even easier to setup.

**Comment by axonasif** [comment]
> @axonasif Thank you for the feedback. I'm not sure I understand the use-case though; instead it strikes me that that approach comes with some drawbacks which make it unfavourable imho:
> - the overlay would affect the entire `$HOME` directory and it's unclear what the meaning of the lowerdir would be then
> - akin to the former point, due to the symlinks changes made to the files can be committed back to the repo, with the overlay that would not be so easy
> - the filesystem setup in a workspace is already dangerously complex (shiftfs, fuse-overlayfs, idmapped mounts in the future), and I fret to add another layer to the mix.

Thank you @csweichel for sharing your thoughts on the drawbacks. Now it makes sense to me why symlinks is a better and probably the only option here, couldn't think of a few of the issues you mentioned.

Also I wanted to ask, do you plan to add a subcommand in the `gp` cli for reapplying the dotfiles? Say I committed some new files in the locally cloned `/home/gitpod/.dotfiles` or in the remote and I want to make Gitpod run the dotfiles setup again.

**Comment by JanKoehnlein** [comment]
@gtsiolis could you revise the visuals?

**Comment by csweichel** [comment]
> I tried this PR with my slightly naive dotfiles repo which just tries to set an alias using [`.bash_profile`](https://github.com/jldec/dotfiles/blob/main/.bash_profile) See https://github.com/jldec/dotfiles
> 
> I noticed from the .dotfiles.log that we are symlinking the `.git` directory (bug) and also the `README.md` file (bug?), but not my `.bash_profile` (maybe because there is already a .bash_profile in which case the log should show the failure)
> 

Great find - I've pushed a change that should prevent this behaviour, i.e. not symlink the `.git` directory.

> 
> Beginner Q: If I just wanted to inject aliases with a dotfile, what would be the best way to do that, without overwriting the existing .bashrc in the workspace? Do I need an install script even for something as basic as this?
> 
> note: GitHub codespaces does a symlink: `.bash_profile -> /workspaces/.codespaces/.persistedshare/dotfiles/.bash_profile` But the .bash_profile does not appear to be run in each codespaces terminal either. I think this is expected because VS Code terminal are not login shells.

Good question. We could make it the default behaviour for dotfiles repo that they overwrite files when symlinking. That would be more in-line with how install scripts likely behave. WDYT?

**Comment by csweichel** [comment]
> Also I wanted to ask, do you plan to add a subcommand in the `gp` cli for reapplying the dotfiles? Say I committed some new files in the locally cloned `/home/gitpod/.dotfiles` or in the remote and I want to make Gitpod run the dotfiles setup again.

That would be handy to have. It's not clear however how that would work
- in case of install scripts because we'd need to assume they're idempotent,
- with the IDE and its environment in case there are changes to the `.bashrc` or `.bash_profile`. Would we restart the IDE to source those changes?

**Comment by axonasif** [comment]
> with the IDE and its environment in case there are changes to the .bashrc or .bash_profile. Would we restart the IDE to source those changes?

I think only informing the user about shell configuration changes is enough and that it will be effective for newly created VSCODE terminals, pre-existing terminals would not be effected unless the user explicitly reload shellrc with `source ~/.bashrc` (or their respective shell specific file). Although it should be possible to autorun such a command on pre-existing terminals, here is an example: https://github.com/osospeed/ttyecho/. While this might not be a neat thing to do.

**Comment by gtsiolis** [comment]
Looking at this now! 👀

File: components/dashboard/src/settings/Preferences.tsx
Comment (gtsiolis) [inline components/dashboard/src/settings/Preferences.tsx:167]: **suggestion**: Looks better as non-abbriviated and more descriptive, no?

```suggestion
                <h4>Repository URL</h4>
```

File: components/dashboard/src/settings/Preferences.tsx
Comment (gtsiolis) [inline components/dashboard/src/settings/Preferences.tsx:168]: **suggestion**: What do you think of adding a placeholder as help text here?

```suggestion
                <input type="text" value={dotfileRepo} onChange={(e) => actuallySetDotfileRepo(e.target.value)} className="w-full" placeholder="e.g. https://github.com/username/dotfiles" />
```

Comment (gtsiolis) [inline components/dashboard/src/settings/Preferences.tsx:168]: **question**: Will this work when using a repository URL without the `.git` suffix?

Comment (gtsiolis) [inline components/dashboard/src/settings/Preferences.tsx:168]: **issue(non-blocking)**: Auto-saving the input here happens in the background but is lacking :a: call to action and :b: some user feedback upon save which both will probably be resolved in the second iteration in https://github.com/gitpod-io/gitpod/issues/7385.

For :a:, we could add an action button below to save changes so that users won't be surprized when they accidentally change this input and dotfiles configuration fails, etc. Alternatively, we could introduce a new auto-save pattern that makes it visible to the user that we auto-save on every key press.

For :b:, ideally this manual save action would also need some feedback in the user interface so that users can safely acknowledge that the setting has been updated. This could be done with a subtle loading indicator (spinner) inside the button when saving changes that remains visible for a noticeable period of time or using a toast notification (https://github.com/gitpod-io/gitpod/issues/3530) which can be considered out of the scope of this PR.

Comment (gtsiolis) [inline components/dashboard/src/settings/Preferences.tsx:168]: **suggestion**: Instead of using the section subtitle to describe this input we could use the already existing pattern of the input help text to make it clear to the user what is expected inside the input.

File: components/dashboard/src/settings/Preferences.tsx
Comment (gtsiolis) [inline components/dashboard/src/settings/Preferences.tsx:165]: **nitpick**: So far, we've generally opted for US English across the dashboard. 🇺🇸 

**question**: Also, what do you think of moving the input help text closer to the input instead of using the section subtitle?

```suggestion
            <p className="text-base text-gray-500 dark:text-gray-400">Customize workspaces using dotfiles.</p>
```

**suggestion**: A link to learn more about how this feature works (conventions, installation, etc) to the docs (https://github.com/gitpod-io/website/issues/1407) once the docs are in place could be also nice, as @jldec mentioned in https://github.com/gitpod-io/gitpod/pull/7337#issuecomment-1002740006.

File: components/dashboard/src/settings/Preferences.tsx
Comment (gtsiolis) [inline components/dashboard/src/settings/Preferences.tsx:166]: **issue(non-blocking)**: The max width of the text input element could be limited as also @jldec mentioned in https://github.com/gitpod-io/gitpod/pull/7337#issuecomment-1002740006. I'd suggest to go with `max-w-md`.

```suggestion
            <div className="mt-4 max-w-md">
```

**Review by gtsiolis — APPROVED** [review]
Looks great, @csweichel! 🔮 

Left some minor comments below that could be addressed in the next iteration. ➿ 

The only non-minor issue relates to the interaction of auto-saving which could be slightly confusing not crystal clear to the user. However, it could be ok to leave this out of the scope of this PR if we'd like to get this in soon and improve in the next iterations. ❗ 

Approving to unblock merging but holding in case we'd like to address any comments regaridng UX below.

| BEFORE | AFTER |
|-|-|
| dotfiles-before | dotfiles-after |

/hold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 2a003b3e07decccf2f17e8bd951d9aaa0564dc09</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7337#pullrequestreview-842709334" title="Approved">gtsiolis</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **filiptronicek** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @filiptronicek` in a comment when ready.

Associated issue: *#5198*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [gtsiolis]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [gtsiolis]
- **[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by csweichel** [comment]
Thanks @gtsiolis for your detailed feedback. I've incorporated it as best I could. There's still some padding that's off, and it's a far cry from your original designs - but a first step :)

**Comment by axonasif** [comment]
@csweichel I'm unable to test this feature 😅 , the test url redirects to such a url: https://cw-dotfiles.staging.gitpod-dev.com/sorry#Login%20was%20interrupted:%20Maximum%20number%20of%20users%20permitted%20by%20the%20license%20exceeded

**Comment by csweichel** [comment]
> @csweichel I'm unable to test this feature 😅 , the test url redirects to such a url: https://cw-dotfiles.staging.gitpod-dev.com/sorry#Login%20was%20interrupted:%20Maximum%20number%20of%20users%20permitted%20by%20the%20license%20exceeded

Too many other folks had tried this before :) I've re-deployed the branch with a clean slate. Once that's through you should be able to try this change.

**Comment by filiptronicek** [comment]
@csweichel I tried authenticating and getting the same redirect URL, basically meaning that my auth popup window never gets closed automatically. Even when trying to auth via GitLab, I can't proceed. Is there any technical limitation I have never encountered or am I just unlucky? 😄

**Comment by axonasif** [comment]
@filiptronicek yeah, it's still the same. I thought it will take some time but no 😅

**Comment by nberlette** [comment]
Just thought I'd chime in on this - when I took this PR for a test drive, the dashboard initially (and erroneously) reported I'd been **demoted** to free tier. It also appeared to wipe out all workspace data. 

I was neither able to get around that obstacle, nor able to even spin up a workspace 🤕... but all my data was right where I left it, once back in the main dashboard. 

Sorry, I know this is probably of no help.. but it felt relevant considering the auth issues you encountered. Thx 🙏

**Comment by JanKoehnlein** [comment]
> @csweichel I tried authenticating and getting the same redirect URL, basically meaning that my auth popup window never gets closed automatically. Even when trying to auth via GitLab, I can't proceed. Is there any technical limitation I have never encountered or am I just unlucky? 😄

Same here

**Comment by csweichel** [comment]
@JanKoehnlein @axonasif @filiptronicek I've wiped all users from this installation. Please try again.

**Comment by axonasif** [comment]
@csweichel I was able to login now, will spin up a workspace and let you know how it goes!

**Review by filiptronicek — COMMENTED** [review]
I have just tried using my dotfiles (https://github.com/filiptronicek/dotfiles) and am getting interesting results:
every time when starting a workspace, the setup screen keeps on refreshing (e.g. `white-wolverine-55vly6wb`)

Update: This seems to be happening even without a dotfiles repo...

File: components/supervisor/pkg/supervisor/supervisor.go
Comment (akosyakov) [inline components/supervisor/pkg/supervisor/supervisor.go:373]: not exactly sure why terminal service is needed here

**Review by akosyakov — APPROVED** [review]
/approve
/hold

I have not tried, but supervisor code looks alright, hold since someone should test that it works.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 7bfb85c69fa5e087d088b71b7c23874c9bd5204c</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7337#pullrequestreview-845098931" title="Approved">akosyakov</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7337#pullrequestreview-842709334" title="Approved">gtsiolis</a>*

Associated issue: *#5198*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [gtsiolis]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [gtsiolis]
- ~~[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)~~ [akosyakov]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by csweichel** [comment]
Haven't heard anything about this not working. Let's merge and iterate on main.

/hold cancel

**Comment by filiptronicek** [comment]
> Haven't heard anything about this not working. Let's merge and iterate on main.

@csweichel https://github.com/gitpod-io/gitpod/pull/7337#pullrequestreview-845026371 did you see or experience this one?

**Comment by JanKoehnlein** [comment]
@filiptronicek It worked for me since https://github.com/gitpod-io/gitpod/pull/7337#issuecomment-1005847118

**Comment by Milo123459** [comment]
What file does it use to boot the dotfiles? Ie, is it like an install.sh?

**Comment by filiptronicek** [comment]
@Milo123459 from the [RFC](https://gitpod.notion.site/Dotfiles-in-Gitpod-workspaces-b46b8723e9fe4efdbede72daa311961f):
> 2. **Installing the dotfiles**: we look for a setup script in the cloned dotfile repo using this list:
    - *install.sh*
    - *install*
    - *bootstrap.sh*
    - *bootstrap*
    - *script/bootstrap*
    - *setup.sh*
    - *setup*
    - *script/setup*
    If no such script is found, we symlink every file and folder starting with a `.` from within the cloned repo to `/home/gitpod`. For example:

**Pull request merged**
